src/fromager/commands/bootstrap.py

```diff
@@ -156,18 +156,21 @@ def bootstrap(
 
     constraints_filename = wkctx.work_dir / "constraints.txt"
     logger.info(f"writing installation dependencies to {constraints_filename}")
-    with open(wkctx.work_dir / "constraints.txt", "w") as f:
-        write_constraints_file(graph=wkctx.dependency_graph, output=f)
+    with open(constraints_filename, "w") as f:
+        if not write_constraints_file(graph=wkctx.dependency_graph, output=f):
+            raise ValueError(
+                f"Could not produce a pip compatible constraints file. Please review {constraints_filename} for more details"
+            )
```

Comment:
If we throw the exception inside the `with` block, is the file still written properly?

Reply:
the writing happens inside the `write_constraints_file` function so it should be fine

Reply:
not sure, know any small enough packages that have conflicting installation time deps?

Reply:
What if we construct our own fake package?

Reply:
I realized we can just pass 2 conflicting toplevel dependencies